lib/import_govdelivery_csv.rb

```diff
@@ -53,56 +49,89 @@ def import_subscribers
     existing_addresses = Subscriber.where(address: addresses).pluck(:address)
 
     puts "Identifying new subscribers..."
+
     new_addresses = addresses - existing_addresses
 
     columns = %w(address)
     records = new_addresses.map { |address| [address] }
 
     puts "Importing records..."
+
     count = Subscriber.import!(columns, records).ids.count
 
     puts "#{count} subscribers imported!"
   end
 
+  def all_subscribers
+    @all_subscribers ||= begin
+      puts "Loading all subscribers from database..."
+      Subscriber.all.each_with_object({}) do |subscriber, hash|
+        hash[subscriber.address] = subscriber
+      end
+    end
+  end
+
   def subscriber_for_row(row)
-    Subscriber.find_by!(address: address_from_row(row))
+    address = address_from_row(row)
+    all_subscribers.fetch(address)
+  end
+
+  def all_subscribables
+    @all_subscribables ||= begin
+      puts "Loading all subscribables from database..."
+      SubscriberList.all.each_with_object({}) do |subscribable, hash|
```

Comment:
Not a biggie but I think you can do this with `SubscriberList.all.index_by(&:gov_delivery_id)`